bb8/src/inner.rs

```diff
@@ -116,10 +126,14 @@ where
             }
         };
 
-        match timeout(self.inner.statics.connection_timeout, future).await {
+        let result = match timeout(self.inner.statics.connection_timeout, future).await {
             Ok(result) => result,
             _ => Err(RunError::TimedOut),
-        }
+        };
+
+        self.pool_inner_stats.record_get(with_contention);
```

Comment:
If this times out, should we still record a get, or is this arguably a third kind of event?

Reply:
IMO we should still record it, timeout is the just a safeguard for not having indefinitely polling for getting a connection, which means that suffered from the issues that this new statistic is trying to inform to.

True that the user that uses this statistic should not blindly believe that an increase of gets with contention is just an issue with the connection allocated in the pool, as you were mentioning timeout could also indicate that the server is for example having issues creating new connections (for whatever reasons).

I would not go for a time out event for now, unless someones believes that the addition of this one would address some lack on observability.

Reply:
Hmm, I think it would be strictly more useful to have three fields on `AtomicStatistics`:

* `get_direct`
* `get_waited`
* `get_timed_out`

Reply:
Oks, lets go for it